internal/configuration/validator/identity_providers.go

```diff
@@ -82,17 +96,96 @@ func validateOIDCClients(configuration *schema.OpenIDConnectConfiguration, valid
 	}
 }
 
+func validateOIDCClientScopes(c int, configuration *schema.OpenIDConnectConfiguration, validator *schema.StructValidator) {
+	if len(configuration.Clients[c].Scopes) == 0 {
+		configuration.Clients[c].Scopes = schema.DefaultOpenIDConnectClientConfiguration.Scopes
+		return
+	}
+
+	if !utils.IsStringInSlice("openid", configuration.Clients[c].Scopes) {
+		configuration.Clients[c].Scopes = append(configuration.Clients[c].Scopes, "openid")
+	}
+
+	for _, scope := range configuration.Clients[c].Scopes {
+		if !utils.IsStringInSlice(scope, validScopes) {
```

Comment:
I think the concept of valid scope is limited to ID tokens but does not apply to access tokens. To me it should be ok to request a custom scope from an app and attach it to the access token. Maybe there is something to consider here regarding this array.

Reply:
I got my answer in the Discovery RFC: `Servers MAY choose not to advertise some supported scope values even when this parameter is used, although those defined in [OpenID.Core] SHOULD be listed, if supported.`
So this support list only relates to OIDC not OAuth2.